### ISSUE TITLE
Add geographically weighted regressions

### ISSUE THREAD

**Issue by rvanmazijk** [rvanmazijk/Cape-vs-SWA#15]
In #11 mentioned the importance of autocorrelative structure in space.

Want to apply GWR for HDS richness ~ avg QDS richness & turnover models, and also (maybe) for richness/turnover ~ environment/heterogeneity models (eventually!).

**Comment by rvanmazijk** [comment]
This is a good thing to prioritise, especially over #16, as this really would make a difference in our results to a greater extent than using mixed models (I think).

In other words, GWR should be in the ms I submit to J Biogeog *before* review.

**Comment by rvanmazijk** [comment]
Update: this is going well! See 0832dffc6d8214f008cbea8b14b23e412fcb0f32

**Comment by rvanmazijk** [comment]
As of #31 (d8f154d1919219d457c794a3ac875c5b7079c52f), I am putting GWR on hold.

It works brilliantly for richness ~ enviro + roughness (using `spgwr`) for each region _separately_

But when it comes to fitting models with a region term (`"GCFR" or `"SWAFR"`), or even a dummy-coded region term (`1` or `0`), it fails. `spgwr` gets NAs in its estimates and can't compute AICc in these cases, and `GWmodel` simply hangs....

**Comment by rvanmazijk** [comment]
I also tried using the `GWmodel::gwr.mixed()`, thinking I could fit a random effect for region.

Turns out, as I realised in fcdb0fbefca39cb7215bc9408b60db21b7a1b599, that this is not a mixed effects model the way I thought. The `fixed.vars` for fixed effects become spatially global. The random effects in this model are then simply those allowed to vary geographically again.

Not a mixed effect model in the sense of grouping error by region.

Maybe spatially groupd information regarding coefficient estiamtes in either region could be useful??
